apps/meteor/app/api/server/lib/maybeMigrateLivechatRoom.ts

```diff
@@ -0,0 +1,26 @@
+import { isOmnichannelRoom, type IRoom } from '@rocket.chat/core-typings';
+import { Rooms } from '@rocket.chat/models';
+import type { FindOptions } from 'mongodb';
+
+import { migrateVisitorIfMissingContact } from '../../../livechat/server/lib/contacts/migrateVisitorIfMissingContact';
+
+export async function maybeMigrateLivechatRoom(room: IRoom | null, options: FindOptions<IRoom> = {}): Promise<IRoom | null> {
```

Comment:
btw this is a bit ugly dont you think?

maybe we should create a new endpoint only for omnichannel